tests/version_tests.cpp

```diff
@@ -0,0 +1,282 @@
+    trace = admincall(from, to, silkworm::Bytes(evmc::bytes32{}), *data, 1000000, evm_account_name);
+    BOOST_REQUIRE(trace->action_traces.size() == 2);
+    BOOST_REQUIRE(trace->action_traces[0].act.account == "evm"_n);
+    BOOST_REQUIRE(trace->action_traces[0].act.name == "admincall"_n);
+    BOOST_REQUIRE(trace->action_traces[1].act.account == "evm"_n);
+    BOOST_REQUIRE(trace->action_traces[1].act.name == "pushtx"_n);
+
+    /////////////////////////////////////
+    /// change EOS EVM VERSION => 1   ///
+    /////////////////////////////////////
+
+    setversion(1, evm_account_name);
+    produce_blocks(2);
+
+    auto get_tx_from_trace = [&](const bytes& v) {
+      auto evmtx_v = fc::raw::unpack<evm_test::evmtx_type>(v.data(), v.size());
+
+      BOOST_REQUIRE(std::holds_alternative<evm_test::evmtx_v0>(evmtx_v));
+
+      const auto& evmtx = std::get<evm_test::evmtx_v0>(evmtx_v);
+      BOOST_REQUIRE(evmtx.eos_evm_version == 1);
+
+      silkworm::Transaction tx;
+      silkworm::ByteView bv{(const uint8_t*)evmtx.rlptx.data(), evmtx.rlptx.size()};
+      silkworm::rlp::decode(bv, tx);
+      BOOST_REQUIRE(bv.empty());
+
+      return tx;
+    };
+
+    // Test traces of `handle_evm_transfer` (EVM VERSION=1)
+    trace = transfer_token("alice"_n, evm_account_name, make_asset(to_bridge), evm1.address_0x());
+    BOOST_REQUIRE(trace->action_traces.size() == 4);
+    BOOST_REQUIRE(trace->action_traces[0].act.account == "eosio.token"_n);
+    BOOST_REQUIRE(trace->action_traces[0].act.name == "transfer"_n);
+    BOOST_REQUIRE(trace->action_traces[1].act.account == "eosio.token"_n);
+    BOOST_REQUIRE(trace->action_traces[1].act.name == "transfer"_n);
+    BOOST_REQUIRE(trace->action_traces[2].act.account == "eosio.token"_n);
+    BOOST_REQUIRE(trace->action_traces[2].act.name == "transfer"_n);
+    BOOST_REQUIRE(trace->action_traces[3].act.account == "evm"_n);
+    BOOST_REQUIRE(trace->action_traces[3].act.name == "evmtx"_n);
+
+    auto tx = get_tx_from_trace(trace->action_traces[3].act.data);
+    BOOST_REQUIRE(tx.value == intx::uint256(balance_and_dust{make_asset(to_bridge), 0}));
+    BOOST_REQUIRE(tx.to == evm1.address);
+
+    // Test traces of `pushtx` (EVM VERSION=1)
+    silkworm::Transaction txin {
+      .type = silkworm::Transaction::Type::kLegacy,
+      .max_priority_fee_per_gas = config.gas_price,
+      .max_fee_per_gas = config.gas_price,
+      .gas_limit = 10'000'000,
+      .to = contract_address,
+      .data = *data
+    };
+
+    evm1.sign(txin);
+    trace = pushtx(txin);
+
+    BOOST_REQUIRE(trace->action_traces.size() == 2);
+    BOOST_REQUIRE(trace->action_traces[0].act.account == "evm"_n);
+    BOOST_REQUIRE(trace->action_traces[0].act.name == "pushtx"_n);
+    BOOST_REQUIRE(trace->action_traces[1].act.account == "evm"_n);
+    BOOST_REQUIRE(trace->action_traces[1].act.name == "evmtx"_n);
+
+    auto txout = get_tx_from_trace(trace->action_traces[1].act.data);
+    BOOST_REQUIRE(txout == txin);
+
```

Comment:
Could add check to retrieve count for `evm1.address_0x()` from deployed contract and ensure it is 1.